src/app/AppToolbarArea.tsx

```diff
@@ -1,83 +1,8 @@
-import { Box } from '@mui/material'
-import { createContext, useContext, useMemo, useState } from 'react'
-import { createPortal } from 'react-dom'
+import createSlotPortal from '#src/lib/createSlotPortal'
 
 export const {
   reactContext: appToolbarContext,
   SlotComponent: AppToolbarSlot,
   PortalComponent: AppToolbarPortal,
   PortalProvider: AppToolbarProvider,
-} = createSlotPortal({
-  renderSlot(props) {
-    return (
-      <Box
-        sx={{ display: 'flex', alignItems: 'center', gap: 2 }}
-        ref={props.ref}
-      />
-    )
-  },
-})
-
-function createSlotPortal<T extends HTMLElement = HTMLElement>(opts: {
```

Comment:
This should have been in its own file from the get go. There's also a change: The `renderSlot` function has gone from something you pass to `createSlotPortal` to a prop on `SlotComponent`. Figured if this does get more use, it would be better in the long run to have it be a prop, where there's more context in case its rendering ever needs to change in the future, compared to a static thing passed to a function.